common/containers/Tabs/SendTransaction/index.jsx

```diff
@@ -381,16 +404,42 @@ export class SendTransaction extends React.Component {
       unit
     });
   };
+
+  generateTx = async () => {
+    const { nodeLib, wallet } = this.props;
+    const address = await wallet.getAddress();
+
+    try {
+      const transaction = await nodeLib.generateTransaction(
+        {
+          to: this.state.to,
+          from: address,
+          value: this.state.value,
+          gasLimit: this.state.gasLimit,
+          gasPrice: this.props.gasPrice,
+          data: this.state.data,
+          chainId: this.props.network.chainId
+        },
+        wallet
+      );
```

Comment:
Already checking valid to address. I'll add those gas checks to the `generateTransaction` errors though, good call.

I'm not sure how one would get here with an invalid from address though, since this uses the wallet class to do everything, so I'm not concerned about that.